src/app/pages/profile/financial-information/financial-information.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { ProfileService } from 'src/app/services/profile/profile.service';
+import { first } from 'rxjs/operators';
+import { removeSubscription } from 'src/app/helpers/unsubscribe';
+
+@Component({
+  selector: 'app-financial-information',
+  templateUrl: './financial-information.component.html',
+  styleUrls: ['financial-information.component.scss']
+})
+export class FinancialInformationComponent implements OnInit, OnDestroy {
+  deposits: any;
+  cardInfo: any;
+  constructor(
+    private profileService: ProfileService,
+  ) {}
+  subscribe: Subscription[] = [];
+  ngOnInit() {
+    this.getDeposits();
+    this.getCardInfo();
+  }
+
+  getDeposits() {
+    this.profileService
+      .getDeposits()
+      .pipe(first())
+      .subscribe(
+        response => {
+          response.data.transactions.map(transaction => {
+            if ( transaction.percentage_completion !== '100' ) {
+              this.deposits = transaction;
+            }
+            return this.deposits;
+          });
+        },
+        error => {
+          this.deposits = {error: 'deposits couldn\'t be found'};
+        }
+      );
+  }
+
+  getCardInfo() {
+    this.profileService.userProfile$.subscribe(response => {
+      const profile = response.data.profile;
+      if (Object.keys(profile.card_info.card_info).length === 0) {
+        this.cardInfo = null;
+      } else {
+        const cardNumberFull = [...profile.card_info.card_info.card_number];
+        this.cardInfo = {
+          cardOwner : `${profile.user.first_name} ${profile.user.last_name}`,
+          cardNumber: `${cardNumberFull[9]} ${cardNumberFull[10]} ${cardNumberFull[11]} ${cardNumberFull[12]}`,
+          cardExpiry: profile.card_info.card_info.card_expiry,
+        };
+        this.getCardBrand(profile.card_info.card_info.card_brand);
+      }
+    });
+  }
+
+  getCardBrand(brand) {
+    const cardUrls = {
+      VISA: 'https://res.cloudinary.com/landville/image/upload/v1566848151/FRONTEND_ASSETS/IMAGES/jrlki6prt2mzkycpw5gs.png',
```

Comment:
Also `VISA` seems to be a variable defined which is not in the same format as the string keys for the rest of the other brands. So why is it defined as `VISA` instead of `'VISA'`

Reply:
Thank you @malep2007 The reason why there are quotes on other keys in this object expect`VISA` is that if you check well other keys are two words and if you don't provide the quotes it will throw an error on the second word on that key, but for `VISA` it is one word and if I add quotes it throws a `tslint` warning saying `unnecessarily quoted property`.

<img width="853" alt="Screen Shot 2019-08-29 at 11 16 14" src="https://user-images.githubusercontent.com/36806576/63930173-17afbc00-ca53-11e9-8c5f-bbc4576ee4db.png">

Reply:
ok